components/tabs/src/schema.rs

```diff
@@ -34,10 +31,15 @@ const CREATE_META_TABLE_SQL: &str = "
 const CREATE_PENDING_REMOTE_DELETE_TABLE_SQL: &str = "
     CREATE TABLE IF NOT EXISTS pending_remote_tab_closures (
         id                      INTEGER PRIMARY KEY AUTOINCREMENT,
-        client_id               TEXT NOT NULL,
-        url                     TEXT NOT NULL,
-        time_requested_close    INTEGER NOT NULL
+        device_id               TEXT NOT NULL,
+        command                 TEXT NOT NULL, -- probably 'close'
+        url                     TEXT,
+        time_requested          INTEGER NOT NULL, -- local timestamp when this was initially written.
+        time_sent               INTEGER -- local timestamp, non-null == no longer pending.
     );
+
+    -- Having an index on a URL isn't ideal, should just hash?
```

Comment:
Hmm, I think it's totally OK to use the URL directly here—a hash (specifically the hashing function that we use for URLs in Places) would prevent us from using a `UNIQUE` index directly; we'd need to make our queries a bit fancier to avoid duplicates.

I think the hash is more about [reducing the on-disk size](https://bugzilla.mozilla.org/show_bug.cgi?id=889561) for tables with lots of rows. With a few hundred or even thousand rows, though, I'm not sure it buys us a ton—counterintuitively, for smaller tables, it actually increases the storage size!